src/notification-banner/private/useNotificationBanners.jsx

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * The updateBannerState ref stores the update state function used to manage the banner rendered in the NotificationBanners component.
+   * This ties the state updates to the `useNotificationBanners` hook, while allowing the NotificationBanners to be rendered above or below
+   * the NotificationBannerProvider.
+   */
+  const updateBannerState = React.useRef();
+
+  const bannerProviderValue = React.useMemo(() => {
+    /**
+     * Adds the banner ID and props to the collection of registered banners. Once registered,
+     * it updates the NotificationBanner's state to render the new Banner.
+     *
+     * @param {UUID} bannerId - unique ID associated to the banner
+     * @param {Object} bannerProps - react props associated to the banner. See ../NotificationBanner's propTypes.
+     */
+    const registerNotificationBanner = (bannerId, bannerProps) => {
+      if (!bannerId) {
+        throw new Error('A banner cannot be registered without an identifier.');
+      }
+
+      const { variant } = bannerProps;
+
+      if (!registeredBanners.current[variant]) {
+        registeredBanners.current[variant] = {};
+      }
+
+      registeredBanners.current[variant][bannerId] = { key: bannerId, ...bannerProps };
+
+      if (updateBannerState.current) {
+        updateBannerState.current({ ...registeredBanners.current });
+      }
+    };
+
+    /**
+     * Removes the banner ID and props from the collection of registered banners. Once unregistered,
+     * it updates the NotificationBanner's state to remove the Banner from the list.
+     *
+     * @param {UUID} bannerId - unique ID associated to the banner
+     * @param {String} bannerVariant - the banner variant to remove banner from
+     */
+    const unregisterNotificationBanner = (bannerId, bannerVariant) => {
+      if (!bannerId || !bannerVariant) {
+        throw new Error('A banner cannot be unregistered without an identifier or banner variant.');
+      }
+
+      if (!registeredBanners.current[bannerVariant][bannerId]) {
+        return;
+      }
+
+      delete registeredBanners.current[bannerVariant][bannerId];
+
+      if (updateBannerState.current) {
+        updateBannerState.current({ ...registeredBanners.current });
+      }
+    };
+
+    return {
+      registerNotificationBanner,
+      unregisterNotificationBanner,
+    };
+  }, []);
+
+  /**
+   * Provides the Banner Registration Context to its children.
+   */
+  const NotificationBannerProvider = ({ children }) => (
+    <BannerRegistrationContext.Provider value={bannerProviderValue}>
+      {children}
+    </BannerRegistrationContext.Provider>
+  );
+
+  NotificationBannerProvider.propTypes = {
+    children: PropTypes.node,
+  };
+
+  /**
+   * Renders a list of prioritized notification banners.
+   */
+  const NotificationBanners = () => {
+    const theme = React.useContext(ThemeContext);
+    const [banners, setBanners] = React.useState([]);
+    updateBannerState.current = setBanners;
```

Comment:
👀  we might want to put some comments here about what this line is doing

Reply:
added: https://github.com/cerner/terra-application/pull/45/commits/1efd4b0fe6254e30f83bc4e54571fd2137a9ae27#diff-7732e149c9945ef93b0d5632b181cef4R111